mlflow/server/js/src/experiment-tracking/components/MetricsPlotPanel.tsx

```diff
@@ -73,50 +105,27 @@ export class MetricsPlotPanel extends React.Component<MetricsPlotPanelProps, Met
     containsInfinities: false,
   };
 
-  displayPopover: any;
   intervalId: any;
-
-  // The fields below are exposed as instance attributes rather than component state so that they
-  // can be updated without triggering a rerender.
-  //
-  // ID of Javascript future (created via setTimeout()) used to trigger legend-click events after a
-  // delay, to allow time for double-click events to occur
-  legendClickTimeout = null;
-  // Time (millis after Unix epoch) since last legend click - if two clicks occur in short
-  // succession, we trigger a double-click event & cancel the pending single-click.
-  prevLegendClickTime = (Math as any).inf;
-
-  // Last curve ID clicked in the legend, used to determine if we're double-clicking on a specific
-  // legend curve
-  lastClickedLegendCurveId = null;
-
-  // Max time interval (in milliseconds) between two successive clicks on the metric plot legend
-  // that constitutes a double-click
-  MAX_DOUBLE_CLICK_INTERVAL_MS = 300;
-
-  // Delay (in ms) between when a user clicks on the metric plot legend & when event-handler logic
-  // (to toggle display of the selected curve on or off) actually fires. Set to a larger value than
-  // MAX_DOUBLE_CLICK_INTERVAL_MS to allow time for the double-click handler to fire before firing
-  // a single-click event.
-  SINGLE_CLICK_EVENT_DELAY_MS = this.MAX_DOUBLE_CLICK_INTERVAL_MS + 10;
+  loadMetricHistoryPromise: Promise<any> | null;
+  xAxisType: string | null;
 
   constructor(props: MetricsPlotPanelProps) {
     super(props);
     this.state = {
       historyRequestIds: [],
-      popoverVisible: false,
-      popoverX: 0,
-      popoverY: 0,
-      popoverRunItems: [],
       focused: true,
       loading: false,
+      fullScreenChart: undefined as
```

Comment:
nit: should we explicitly define component's state TS type instead of using `as` here?